src/bsoncxx/document/value.hpp

```diff
@@ -76,6 +81,20 @@ class BSONCXX_API value {
     value(value&&) noexcept = default;
     value& operator=(value&&) noexcept = default;
 
+    // Serializer functions
+    template <typename T,
+              typename std::enable_if<!std::is_same<T, typename array::view>::value, int>::type = 0>
+    explicit value(const T& user_object)
+        : _data(new std::uint8_t[static_cast<std::size_t>(5)],
```

Comment:
This could be replaced with a delegating constructor that takes an empty document. So:
```
explicit value(const T& user_object) : value({}) {
```
That way we don't duplicate the logic of the constructor inline 76.